sycl/source/detail/graph_impl.cpp

```diff
@@ -140,6 +165,40 @@ std::shared_ptr<node_impl> graph_impl::addSubgraphNodes(
   return this->add(Outputs);
 }
 
+std::shared_ptr<node_impl> graph_impl::duplicateAndAddSubgraphNodes(
+    const std::list<std::shared_ptr<node_impl>> &NodeList) {
+  std::map<std::shared_ptr<node_impl>, std::shared_ptr<node_impl>> NodesMap;
+  std::list<std::shared_ptr<node_impl>> NewNodeList;
+
+  for (std::list<std::shared_ptr<node_impl>>::const_iterator NodeIt =
+           NodeList.end();
+       NodeIt != NodeList.begin();) {
+    --NodeIt;
+    auto Node = *NodeIt;
+    std::shared_ptr<node_impl> NodeCopy;
+    duplicateNode(Node, NodeCopy);
+    NewNodeList.push_back(NodeCopy);
+    NodesMap.insert({Node, NodeCopy});
+    for (auto &NextNode : Node->MSuccessors) {
+      if (NodesMap.find(NextNode) != NodesMap.end()) {
+        auto Successor = NodesMap[NextNode];
+        NodeCopy->registerSuccessor(Successor, NodeCopy);
+      }
+    }
+  }
+
+  return addSubgraphNodes(NewNodeList);
+}
+
+std::shared_ptr<node_impl> graph_impl::addSubgraphNodes(
+    const std::shared_ptr<exec_graph_impl> &SubGraphExec,
+    const bool DuplicateNodes) {
```

Comment:
Is this parameter needed? It doesn't seem to be used anywhere.

Reply:
Not sure to understand your comment. Both parameters are used in the function below. 
DuplicateNodes is used to select which types of subgraph addition we want to proceed.

Reply:
I refactorized the code to remove this parameter.